ikarus/python/finiteelements/registerelement.hh

```diff
@@ -120,17 +120,35 @@ void registerElement(pybind11::handle scope, pybind11::class_<FE, options...> cl
 
   if constexpr (requires { std::declval<FE>().materialTangent(); })
     cls.def("materialTangent", [](FE& self) { return self.materialTangent(); });
+}
 
+template <bool defaultInitializers = true, class FE, class... options>
+void registerCalculateAt(pybind11::handle scope, pybind11::class_<FE, options...> cls, auto restultTypesTuple) {
+  using Traits         = typename FE::Traits;
+  using FERequirements = typename FE::FERequirementType;
   cls.def(
       "calculateAt",
-      [](FE& self, const FERequirements& req, const Dune::FieldVector<double, Traits::mydim>& local,
-         ResultType resType) {
-        if (resType == ResultType::linearStress)
-          return self.template calculateAt<ResultType::linearStress>(req, local);
-        else
-          DUNE_THROW(Dune::NotImplemented, "Linear-lastic element only supports linearStress as result.");
+      [&](FE& self, const FERequirements& req, const Dune::FieldVector<double, Traits::mydim>& local,
+          std::string resType) {
+        Eigen::VectorXd result;
+        bool success = false;
+        Dune::Hybrid::forEach(restultTypesTuple, [&](auto i) {
+          using RT = typename std::invoke_result_t<decltype(i)>;
```

Comment:
What is `decltype(i)` returning that was wrong? i would have assumed `std::remove_cvref_t<decltype(i)>` would be enough? Why the strange construction with invoke?

Reply:
I don't know why, but is a function pointer returning the resulttype. 
``` cpp
Ikarus::ResultType::linearStress<double, 1, 1> (*)()
```